### PR TITLE
doma: Move android from lower yocto to upper

### PR DESCRIPTION
**Pull request by rshym**
This PR complements xen-troops/meta-xt-images#197.

### CODE REVIEW

**Comment by rshym** [comment]
This PR needs
xen-troops/xt-distro#53
and
https://github.com/xen-troops/meta-xt-images/pull/197
for proper work.

File: recipes-domu/domu-image-android/domu-image-android.bbappend [hunk 1]
```diff
@@ -1,28 +1,24 @@
...
 
-XT_BB_LAYERS_FILE = "meta-xt-prod-extra/doc/bblayers.conf.domu-image-android"
-XT_BB_LOCAL_CONF_FILE = "meta-xt-prod-extra/doc/local.conf.domu-image-android"
+# For proper build of graphics we need to set proper SOC. This info is available
+# for us from local.conf and we do not need to run do_domd_install_machine_overrides.
+# H3 ES3
+SOC_FAMILY_r8a7795 = "r8a7795"
+SOC_FAMILY_r8a7795-es3 = "r8a7795"
+# M3
+#SOC_FAMILY_r8a7796 = "r8a7796"
```
Comment (iusyk) [inline recipes-domu/domu-image-android/domu-image-android.bbappend:16]: why SOC_FAMILY_r8a7796 = "r8a7796" is commented ?

File: recipes-domu/domu-image-android/domu-image-android.bbappend [hunk 2]
```diff
@@ -1,28 +1,24 @@
...
-"
-
-# these layers will be added to bblayers.conf on do_configure
-XT_QUIRK_BB_ADD_LAYER += "meta-xt-prod-extra"
-
-XT_BB_IMAGE_TARGET = "android"
+# If SOC_REVISION is not set, then ES3 is used as default.
+# Pay atention that ES2 is not supported anymore.
+SOC_REVISION_r8a7795-es3 = ""
+#SOC_REVISION_r8a7795-es2 = "es2"
```
Comment (iusyk) [inline recipes-domu/domu-image-android/domu-image-android.bbappend:23]: if it is not supported - maybe reasonable to remove it ?

File: recipes-domu/domu-image-android/domu-image-android.bbappend (hunk 1 above)
Comment (rshym) [reply recipes-domu/domu-image-android/domu-image-android.bbappend]: Disabled code is removed now. Thanks for pointing.

File: recipes-domu/domu-image-android/domu-image-android.bbappend (hunk 2 above)
Comment (rshym) [reply recipes-domu/domu-image-android/domu-image-android.bbappend]: Removed. Thanks for pointing.

**Comment by rshym** [comment]
Check for supported SOC is added.
If this is not H3 ES3, then error message is printed and build of domu-image-android is failed.
This is done to avoid possible incorrect build of hardware dependent modules.

**Comment by rshym** [comment]
Testing is ongoing.

**Comment by rshym** [comment]
Check of supported SOC tested successfully.
If machine is not H3 ES3, then build of domu--image-android fails with message like this:

ERROR: /mnt/data1t/work_android_reworked/build/meta-xt-images/recipes-domu/domu-image-android/domu-image-android.bb: Android can be built for H3 (r8a7795) ES3 only. Your MACHINEOVERRIDES: rcar:rcar-gen3:r8a7795-es2:salvator-xs-h3.

**Comment by rshym** [comment]
Rebased onto latest master.

**Review by lorc — APPROVED** [review]
`Reviewed-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>`

File: recipes-domu/domu-image-android/domu-image-android.bbappend [hunk 3]
```diff
@@ -1,30 +1,20 @@
...
-# these layers will be added to bblayers.conf on do_configure
-XT_QUIRK_BB_ADD_LAYER += " \
-    meta-xt-prod-extra \
-    meta-xt-prod-domx \
-"
-
-XT_BB_IMAGE_TARGET = "android"
+# For proper build of graphics we need to set proper SOC.
+# For now android is supported only for H3 ES3 (r8a7795)
+SOC_FAMILY = "r8a7795"
```
Comment (andr2000) [inline recipes-domu/domu-image-android/domu-image-android.bbappend:16]: I think even for now we should think about the future
Why not `?=` at least?

File: recipes-domu/domu-image-android/domu-image-android.bbappend [hunk 4]
```diff
@@ -1,30 +1,20 @@
...
+# We use SRC_URI provided by meta-xt-images
 
-SRC_URI = " \
-    repo://github.com/xen-troops/manifests;protocol=https;branch=master;manifest=prod_devel/domu_android_host_tools.xml;scmdata=keep \
-"
+# Android itself doesn't care about hardware, but graphics does.
+# That's why we need to check that we are building for supported hardware.
+python __anonymous () {
+    machines = d.getVar("MACHINEOVERRIDES", expand=True)
+    if ("r8a7795-es3" not in machines):
```
Comment (andr2000) [inline recipes-domu/domu-image-android/domu-image-android.bbappend:10]: So, are we going to extend this list over the time adding new supported platforms then?

File: recipes-domu/domu-image-android/domu-image-android.bbappend [hunk 5]
```diff
@@ -1,30 +1,20 @@
...
-"
-
-XT_BB_IMAGE_TARGET = "android"
+# For proper build of graphics we need to set proper SOC.
+# For now android is supported only for H3 ES3 (r8a7795)
+SOC_FAMILY = "r8a7795"
 
+# Revision need to be set to empty string.
+# In this case graphic binaries will be built for ES3.
+SOC_REVISION = ""
```
Comment (andr2000) [inline recipes-domu/domu-image-android/domu-image-android.bbappend:20]: Same as above

File: recipes-domu/domu-image-android/domu-image-android.bbappend (hunk 3 above)
Comment (rshym) [reply recipes-domu/domu-image-android/domu-image-android.bbappend:16]: Reason is that we support android only on H3 ES3 for now.
Weak default assignment is used in main recipe in xt-images
https://github.com/xen-troops/meta-xt-images/pull/197/commits/102ee8bedb0b1389cfcc68d8b9e972243b838e98#diff-40e21da8563c8353e3008858ea6a70be5b0796c6ee4b209d39aa7348f213bd1aR54

But here in product's recipe, I think it good to have '=' assignment, as only supported value is used.

File: recipes-domu/domu-image-android/domu-image-android.bbappend (hunk 4 above)
Comment (rshym) [reply recipes-domu/domu-image-android/domu-image-android.bbappend:10]: Only if we will provide support for new platforms.

File: recipes-domu/domu-image-android/domu-image-android.bbappend (hunk 5 above)
Comment (rshym) [reply recipes-domu/domu-image-android/domu-image-android.bbappend:20]: Unfortunately, for now we support only H3 ES3 for android. That why I provide hard-coded values for SOC_FAMILY and SOC_REVISON.

**Comment by rshym** [comment]
PR reworked.
Now we have list of supported hardware, and check it against MACHINEOVERRIDES. If no match was found then we print warning to user that build may not work properly.
Also I have returned machine dependent overrides for SOC_FAMILY and SOC_REVISION in order to enable build of gfx and op-tee.

Please review reworked PR.

**Pull request merged**